### PR TITLE
d3d: wrap surfaces in textures

### PR DESCRIPTION
**Pull request by LukeUsher**
Fixes when titles attempt to pass a Surface object to SetTexture,

### CODE REVIEW

File: src/core/hle/D3D8/Direct3D9/Direct3D9.cpp [hunk 1]
```diff
@@ -6943,34 +6960,21 @@ IDirect3DBaseTexture* CxbxConvertXboxSurfaceToHostTexture(xbox::X_D3DBaseTexture
...
+	if (!pHostSurface) {
+        LOG_TEST_CASE("Failed to get host surface");
+		return nullptr;
 	}
 
-	IDirect3DSurface* pHostTextureSurface = nullptr;
-	hRet = pNewHostTexture->GetSurfaceLevel(/*Level=*/0, &pHostTextureSurface);
-	DEBUG_D3DRESULT(hRet, "pHostBaseTexture->GetSurfaceLevel");
+	IDirect3DTexture* pNewHostTexture = nullptr;
+	auto hRet = pHostSurface->GetContainer(__uuidof(IDirect3DTexture9), (void**)&pNewHostTexture);
```
Comment (CookiePLMonster) [inline src/core/hle/D3D8/Direct3D9/Direct3D9.cpp:6970]: `pNewHostTexture` could be of type `IDirect3DBaseTexture` and you may use a macro to make it more concise:
`auto hRet = pHostSurface->GetContainer(IID_PPV_ARGS(&pNewHostTexture));`

Once this is done, a typecast in the return statement can be removed.

**Comment by CookiePLMonster** [comment]
Tested a few games showing the `ActiveTexture set to a surface (non-texture) resource` test case previously:

* **Burnout 3** - No change, as expected
* **Midnight Club 3** - Menu renders properly now, gets ingame with glitched graphics
image
image
* **OutRun 2** - No change, road still doesn't render

**Comment by medievil1** [comment]
almost fixes red dead redemption (TT-026 V1.01) it still has some missing textures though
image

**Comment by Margen67** [comment]
Fixes bloom in Battle for Bikini Bottom

**Comment by LanHikariDS** [comment]
image
Improves the rain in Grand Theft Auto III

image
And also now renders (incredibly broken) 3D views during Steel Battalion: Line of Contact's demo loop

File: src/core/hle/D3D8/Direct3D9/Direct3D9.cpp [hunk 2]
```diff
@@ -5710,9 +5716,20 @@ void CreateHostResource(xbox::X_D3DResource *pResource, DWORD D3DUsage, int iTex
 				DEBUG_D3DRESULT(hRet, "g_pD3DDevice->CreateDepthStencilSurface");
 			}
 			else {
-				D3DPool = D3DPOOL_SYSTEMMEM;
-				hRet = g_pD3DDevice->CreateOffscreenPlainSurface(dwWidth, dwHeight, PCFormat, D3DPool, &pNewHostSurface, nullptr);
-				DEBUG_D3DRESULT(hRet, "g_pD3DDevice->CreateOffscreenPlainSurface");
+				hRet = g_pD3DDevice->CreateTexture(dwWidth * g_RenderScaleFactor, dwHeight * g_RenderScaleFactor, 1, D3DUSAGE_RENDERTARGET,
```
Comment (PatrickvL) [inline src/core/hle/D3D8/Direct3D9/Direct3D9.cpp:5719]: It appears like this statement blocks is identical to the D3DUSAGE_RENDERTARGET case above.
Maybe simply remove that?

Also, it'd be nice to add a comment explaining why this approach is used, as opposed to what we had before

**Comment by escape209** [comment]
Partially fixes the distortion overlay in Battle for Bikini Bottom. The screen (minus the HUD) would previously be black for the duration of the effect.

image

File: src/core/hle/D3D8/Direct3D9/Direct3D9.cpp (hunk 1 above)
Comment (PatrickvL) [reply src/core/hle/D3D8/Direct3D9/Direct3D9.cpp]: Done

**Review by PatrickvL — APPROVED** [review]

**Pull request merged**